tests/services/pids/test_pids_tasks.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN
+#
+# Invenio-RDM-Records is free software; you can redistribute it
+# and/or modify it under the terms of the MIT License; see LICENSE file for
+# more details.
+
+"""PID service tasks tests."""
+
+
+from collections import namedtuple
+
+import pytest
+from invenio_pidstore.models import PIDStatus
+
+from invenio_rdm_records.proxies import current_rdm_records
+
+RunningApp = namedtuple("RunningApp", [
+    "app", "location", "superuser_identity", "resource_type_v",
+    "subject_v", "languages_v", "title_type_v"
+])
+
+
+@pytest.fixture
+def running_app(
+    app, location, superuser_identity, resource_type_v, subject_v, languages_v,
+        title_type_v):
+    """This fixture provides an app with the typically needed db data loaded.
+
+    All of these fixtures are often needed together, so collecting them
+    under a semantic umbrella makes sense.
+    """
+    return RunningApp(app, location, superuser_identity,
+                      resource_type_v, subject_v, languages_v, title_type_v)
+
+
+def assert_pids_are_registered(dict_record):
+    """Asserts that pids status are registered."""
+    service = current_rdm_records.records_service
+    for scheme, value in dict_record["pids"].items():
+        provider = service.pids.get_provider(scheme, value["provider"])
+        pid = provider.get(dict_record["pids"][scheme]["identifier"])
+        assert pid.status == PIDStatus.REGISTERED
+
+
+def test_publish_record(running_app, es_clear, minimal_record):
+    """No pid provided, creating one by default."""
+    service = current_rdm_records.records_service
+    superuser_identity = running_app.superuser_identity
+    draft = service.create(superuser_identity, minimal_record)
+    record = service.publish(draft.id, superuser_identity)
+    dict_record = record.to_dict()
```

Comment:
is this needed? cant you do `record["pids"]` directly?  e.g. https://github.com/inveniosoftware/invenio-rdm-records/pull/814/files#diff-4d69e1b21e25061eb03ca154d0f090414925e0529fede5b88a2b6044d8d9e913R41